app/src/app/layout.tsx

```diff
@@ -3,14 +3,55 @@ import {Inter} from 'next/font/google';
 import './globals.css';
 import {Theme} from '@radix-ui/themes';
 import '@radix-ui/themes/styles.css';
+import {getDocument} from '@/app/utils/api/apiHandlers/getDocument';
+import {isArray} from 'lodash';
 
 const inter = Inter({subsets: ['latin']});
 
-export const metadata: Metadata = {
-  title: 'Districtr 2.0',
-  description: 'Districtr reboot',
+type MetadataProps = {
+  searchParams: Promise<{document_id?: string | string[] | undefined}>;
 };
 
+const DISTRICTR_LOGO = [
+  {
+    url: 'https://districtr-v2-frontend.fly.dev/_next/image?url=%2Fdistrictr_logo.jpg&w=1920&q=75',
+    width: 1136,
+    height: 423,
+  },
+];
+
+export async function generateMetadata({searchParams}: MetadataProps): Promise<Metadata> {
+  const resolvedParams = await searchParams;
+  const document_id = resolvedParams?.document_id ?? '';
+  const singularDocumentId = isArray(document_id) ? document_id[0] : document_id;
```

Comment:
Nit: I'd like to phase out the lodash dependency at some point, `Array.isArray` should do the trick here

Reply:
Makes sense 👍